### DISCUSSION TITLE
Multiple forms in a online workflow behind different types of authentications

### DISCUSSION THREAD

**Discussion by gl2007** [firecrawl/firecrawl#696] [Q&A]
Is the library able to authenticate with websites given the credentials and go through a workflow to scrape data from each form in the workflow? An example would be like tax forms workflows on websites. These different websites might have different kinds of authentications.

Have not looked at this library in detail, hence asking before I dive deep.

**Comment by 173787247** [comment]
## Files / crawl / PDF («Multiple forms in a online workflow behind different types of authentications»)

- **PDF + CJK**: need an embeddable font / a backend that keeps unicode; “garbled markdown” is usually extraction encoding, not your editor.
- **Crawl**: respect robots, cap depth, and don’t dump full HTML into the LLM — markdown + metadata, or you blow the context.
- **Memory**: Playwright/Chromium crawlers idle-high; reuse one browser, don’t spawn per URL.
